client/modules/driver/reducers/route.js

```diff
@@ -83,7 +83,11 @@ export const moveWaypoint = (waypoint, idx, getWaypoints) => (dispatch, getState
   // )
 }
 
-export const requestGoogleRoute = (waypoints, optimize) => dispatch => {
+export const requestGoogleRoute = (waypoints, optimize) => (dispatch, getState) => {
+  // TODO: this should be using selectors, but the needed ones are defined here
+  if (getState().delivery.route.fetching)
+    return Promise.reject(new TypeError('Already requesting a route'))
```

Comment:
It's not a type error, why not just Error?

Reply:
@jspaine Changed it.

Reply:
Thanks!